src/utils/pattern.ts

```diff
@@ -184,3 +184,35 @@ export function matchAny(entry: string, patternsRe: PatternRe[]): boolean {
 
 	return patternsRe.some((patternRe) => patternRe.test(filepath));
 }
+
+export function matchAnyWithLevel(setOfGroups: PatternFloatingGroupOfSegments[][], part: string, level: number): boolean {
+	for (const [firstGroup, ...groups] of setOfGroups) {
+		/**
+		 * In this case, the pattern has a globstar and we must read all directories unconditionally.
+		*/
+		if (groups.length !== 0) {
```

Comment:
Check the `{blocks-*,construct}/**/*.css` pattern. In this case we does not trying to match `{blocks-*,construct}`.